help_docs: Document Markdown stream/topic auto-linking feature.

laurynmm opened the pull request:
Adds a section about the Zulip Markdown feature to auto-link to existing streams and topics in a Zulip message to the existing help center documentation for [linking to a message or conversation](https://zulip.com/help/link-to-a-message-or-conversation). Also, does a little reformatting of existing text/tabs in that article and adds a related links section to the end of the page.

The [Markdown formatting](https://zulip.com/help/format-your-message-using-markdown) help center documentation already covers the majority of this information. We could potentially add the "tip" information (re: the menus that appear while typing) to that page as well if it seems appropriate, but I didn't feel like that was necessary.

Fixes #21085.

**Screenshot of updated "Link to a message or conversation" page:**
![Screenshot from 2022-02-17 13-10-32](https://user-images.githubusercontent.com/63245456/154480202-f40d0a3b-42db-43ea-bb7d-54ef9226fbbf.png)

**Screenshot of minor update to "Markdown formmating" page:**
![Screenshot from 2022-02-17 13-34-02](https://user-images.githubusercontent.com/63245456/154482883-1e5b4b9b-0a50-4878-a659-8cf492612c6c.png)

timabbott commented on templates/zerver/help/format-your-message-using-markdown.md:
```diff
@@ -62,8 +62,8 @@ Numbered lists
 
-Zulip auto-linkifies URLs and valid stream names. You can also add a
-[custom linkifier](/help/add-a-custom-linkifier) to link
+Zulip auto-linkifies URLs and valid stream (and topic) names. You can also
```
I think maybe it'd be work having this link to the page about linking to a conversation.

timabbott commented on templates/zerver/help/link-to-a-message-or-conversation.md:
```diff
@@ -1,10 +1,29 @@
+    the desired stream name from the menu, you can press `>` for a list
+    of valid topic names in that particular stream.
+
+## Permanent link to a stream or topic
```
I think this should just be "Link to..." -- the links this generates are permanent for a stream but not for a topic.

laurynmm replied:
If that's the case, then I think we should also adjust the first sentence of the page as well for consistency ...
"Share links to messages, topics, and streams." vs "Share permanent links to messages, topics, and streams."

The description text for links to specific messages explicitly says that it generates a permanent link.

laurynmm commented:
I've updated for @timabbott 's review comments, with one small addition of removing "permanent" from the first sentence on the page. Updated screenshots are below.

@alya - do you want to look over these updates as well?

One thing I noticed is that there's also a similar way to create a permanent link for topics with the three-dots menu in the left sidebar, which is not documented here. If we want to add that information to the doc, then it could be a follow-up PR for review purposes.

**Updated screenshots**:
![Screenshot from 2022-02-21 12-30-22](https://user-images.githubusercontent.com/63245456/154952191-20296919-9ddf-4eb2-be12-6817e9428ef6.png)
![Screenshot from 2022-02-21 12-30-50](https://user-images.githubusercontent.com/63245456/154952195-7db3b152-a0e1-4a5b-a6d9-a71357c87c29.png)

alya commented:
Thanks for working on this! I posted a revised version at #21210 -- feedback is appreciated. :)

alya commented:
> One thing I noticed is that there's also a similar way to create a permanent link for topics with the three-dots menu in the left sidebar, which is not documented here. If we want to add that information to the doc, then it could be a follow-up PR for review purposes.

Yes, a PR for this would be great! The URL bar approach doesn't work in the desktop app, so making this change would be very helpful.  Could you also describe the method of right-clicking on a stream name in the left sidebar, which is how you can get a stream link from the desktop app?

laurynmm commented:
> Yes, a PR for this would be great! The URL bar approach doesn't work in the desktop app, so making this change would be very helpful. Could you also describe the method of right-clicking on a stream name in the left sidebar, which is how you can get a stream link from the desktop app?

Sounds good. I just downloaded the desktop app for Linux and the right-click in the left sidebar isn't working ... I have a Mac as well, so I'll download / check there as well before starting a CZO thread about it.

Also, I'll do these updates as a completely separate PR because 1) I think it would be good to go ahead and merge these changes since the auto-linking isn't documented and 2) I don't know (yet) how you create a PR based on another open PR. Do you get the PR branch locally and then add a new commit with your changed to it?

timabbott commented:
Merged the revision on this in #21210.  Thanks @laurynmm!
